x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporter.java

```diff
@@ -689,6 +694,29 @@ private static void configureTimeouts(final RestClientBuilder builder, final Con
         builder.setRequestConfigCallback(new TimeoutRequestConfigCallback(connectTimeout, socketTimeout));
     }
 
+
+    /**
+     * Caches secure settings for use when dynamically configuring HTTP exporters
+     * @param settings settings used for configuring HTTP exporter
+     * @return names of HTTP exporters whose secure settings changed, if any
+     */
+    public static List<String> loadSettings(Settings settings) {
+        final List<String> changedExporters = new ArrayList<>();
+        for (final String namespace : SECURE_AUTH_PASSWORD_SETTING.getNamespaces(settings)) {
+            final Setting<?> s = SECURE_AUTH_PASSWORD_SETTING.getConcreteSettingForNamespace(namespace);
+            final String password = s.get(settings).toString();
+            String existingPassword;
+            synchronized (SECURE_AUTH_PASSWORDS) {
```

Comment:
I tried using `ConcurrentHashMap` to avoid explicit locking but `ConcurrentHashMap::replace()` works only if there is an existing entry for the specified key. I need the K-V entry to be set whether there was an existing entry or not.

Reply:
Simple `ConcurrentHashMap::put()` should work here - it does return previous value and sets entry whether there was one before or not

Reply:
Ah, right, that will work here.